src/cli/telemetry/telemetry.ts

```diff
@@ -170,20 +175,67 @@ async function getInstalledPackages(sys: d.CompilerSystem, config: d.Config): Pr
       ([k]) => k.startsWith('@stencil/') || k.startsWith('@ionic/') || k.startsWith('@capacitor/')
     );
 
-    packages = packageLockJson
-      ? ionicPackages.map(
-          ([k, v]) =>
-            `${k}@${packageLockJson?.dependencies[k]?.version ?? packageLockJson?.devDependencies[k]?.version ?? v}`
-        )
-      : ionicPackages.map(([k, v]) => `${k}@${v}`);
+    try {
+      if (!isUsingYarn(sys)) {
+        packages = await npmPackages(sys, ionicPackages);
+      } else if (isUsingYarn(sys)) {
+        packages = await yarnPackages(sys, ionicPackages);
+      } else {
```

Comment:
Doesn't `isUsingYarn` always return a boolean? I'm not sure I see the scenario where this `else` block is entered (since we test both `isUsingYarn` in the `else if` condition and it's negation in the `if` condition clause)

In fact, I think this could be refactored as such
```
// NOTE to not be committed: I pulled this out to keep our `try` block tiny
const isUsingYarn = isUsingYarn(sys);
try {
  packages = isUsingYarn ? await yarnPackages(sys, ionicPackages) : await npmPackages(sys, ionicPackages);
} catch (e) {
   packages = ionicPackages.map(([k, v]) => `${k}@${v.replace('^', '')}`);
}